src/solver/conv_direct_naive_conv.cpp

```diff
@@ -78,29 +126,45 @@ std::string ConvDirectNaiveConvKernelName(const ConvolutionContext& ctx)
     else
         MIOPEN_THROW("unsupported tensor layout");
 
-    if(ctx.IsFp32())
-        kernel_name << "fp32";
-    else if(ctx.IsFp16())
-        kernel_name << "fp16";
-    else if(ctx.IsBfp16())
-        kernel_name << "bf16";
+    if(IsInputFp32(ctx))
+        kernel_name << "float_";
+    else if(IsInputFp16(ctx))
+        kernel_name << "half_";
+    else if(IsInputBfp16(ctx))
+        kernel_name << "ushort_";
+    else if(IsInputInt8(ctx))
+        kernel_name << "int8_t_";
     else
         MIOPEN_THROW("unsupported data type:");
 
-    return kernel_name.str();
-}
+    if(IsAccInt32(ctx))
+        kernel_name << "int32_t_";
+    else if(IsAccFp64(ctx))
+        kernel_name << "double_";
+    else
+        MIOPEN_THROW("unsupported data type:");
 
-std::string ConvDirectNaiveConvKernelFile(const ConvolutionContext& ctx)
-{
-    if(ConvDirectNaiveConvIsAssemblyKernel(ctx))
-        return "naive_conv_gcn.s";
+    if(IsOutputFp32(ctx))
+        kernel_name << "float";
+    else if(IsOutputFp16(ctx))
+        kernel_name << "half";
+    else if(IsOutputBfp16(ctx))
+        kernel_name << "ushort";
+    else if(IsOutputInt8(ctx))
+        kernel_name << "int8_t";
+    else if(IsOutputInt32(ctx))
+        kernel_name << "int32_t";
     else
-        return "naive_conv.cpp";
+        MIOPEN_THROW("unsupported data type:");
+
+    return kernel_name.str();
 }
 
+std::string ConvDirectNaiveConvKernelFile() { return "naive_conv.cpp"; }
```

Comment:
@aska-0096 @jerryyin @carlushuang @DrizztDoUrden This disables naive assembly kernel without any notice.

[Notice] This way of development is not acceptable.

Reply:
The code has been updated to the following. The ASM kernel seems to be not applicable? @JehandadKhan 
```
std::string ConvDirectNaiveConvKernelFile(const ConvolutionContext& ctx,
                                          const ProblemDescription& problem)
{
    const auto device_name = ctx.GetStream().GetDeviceName();
    // The above function, ConvDirectNaiveConvKernelName is not in sync for the asm kernel,
    // resulting in empty code objects. This happens for systems with COv3 as the default type.
    // if(device_name == "gfx906" || device_name == "gfx908")
    // {
    //     if(ctx.rmv.IsV3() && problem.IsLayoutDefault() && !problem.IsFp8() &&
    //        !problem.IsTensorsCasted() && !problem.IsBfp8())
    //         return "naive_conv_gcn.s";
    // }
    if(problem.IsFp8() || problem.IsTensorsCasted() || problem.IsBfp8())
        return "fp8_naive_conv.cpp";
    return "naive_conv.cpp";
}
```